### PR TITLE
refactor(components): Update switch for retheme [JOB-84553]

### PR DESCRIPTION
**Pull request by chris-at-jobber**
## Motivations

Allow Switch to be re-styled when a given class is present

### Changed

- Enable Switch to be re-themed and create alternate styling for re-theme
- For both modes, set focus to be on `:focus-visible` to follow browser heuristics on when to show focus ring

## Testing

Apply the `jobber-retheme` class to the Storybook iframe `body`

- [ ] With `jobber-retheme`, should look different
- [ ] Without `jobber-retheme`, should look the same

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

File: packages/components/src/Switch/Switch.css [hunk 1]
```diff
@@ -23,7 +27,16 @@
   appearance: none;
 }
 
-.track:focus {
+:global(.jobber-retheme) .track {
+  width: var(--space-largest);
+  height: var(--space-large);
+  border-color: var(--color-interactive);
```
Comment (darryltec) [inline packages/components/src/Switch/Switch.css:33]: As per Slack, this interactive border is only for hover. Should be grey when unchecked

File: packages/components/src/Switch/Switch.css [hunk 2]
```diff
@@ -65,25 +90,42 @@
...
 }
 
+/* Moving this up overwrites disabled styling */
+/* stylelint-disable-next-line no-descending-specificity */
 .disabled .pip {
   border-color: var(--color-disabled--secondary);
 }
+
+.disabled,
+:global(.jobber-retheme) .disabled {
```
Comment (darryltec) [inline packages/components/src/Switch/Switch.css:127]: I believe disabled have a different look when it is unchecked and disabled. For example, SG1

image

This CSS does this

image

**Review by darryltec — APPROVED** [review]
Looks good! Tho this doesn't show a focus ring like SG1 does 
image

**Comment by chris-at-jobber** [comment]
@darryltec did you try tabbing?

**Comment by darryltec** [comment]
Oh, nvm shows up now. needed another refresh haha

**Pull request merged**